siliconcompiler/core.py

```diff
@@ -4034,7 +4034,9 @@ def _reset_flow_nodes(self, flow, steplist, indexlist):
                     for metric in self.getkeys('metric'):
                         self._clear_metric(step, index, metric)
                     for record in self.getkeys('record'):
-                        self._clear_record(step, index, record)
+                        # Job-wide record values may need to be preserved across runs.
+                        if self.get('record', record, field='pernode') != 'required':
```

Comment:
Are there other records than the one required for `resume` that might be set to `required`?

Reply:
I don't think so, but making a one-off exception for that record felt sort of messy. I'm definitely open to other ways of avoiding clearing this record if you have ideas.

Reply:
maybe make an optional parameter to `_clear_record` to preserve the remoteid?